tests/components/deconz/test_sensor.py

```diff
@@ -185,87 +171,35 @@ async def test_allow_clip_sensors(hass):
-
-    presence_sensor = hass.states.get("sensor.presence_sensor")
-    assert presence_sensor is None
-
-    switch_1 = hass.states.get("sensor.switch_1")
-    assert switch_1 is None
-
-    switch_1_battery_level = hass.states.get("sensor.switch_1_battery_level")
-    assert switch_1_battery_level is None
 
-    switch_2 = hass.states.get("sensor.switch_2")
-    assert switch_2 is None
-
-    switch_2_battery_level = hass.states.get("sensor.switch_2_battery_level")
-    assert switch_2_battery_level.state == "100"
-
-    daylight_sensor = hass.states.get("sensor.daylight_sensor")
-    assert daylight_sensor is None
-
-    power_sensor = hass.states.get("sensor.power_sensor")
-    assert power_sensor.state == "6"
-
-    consumption_sensor = hass.states.get("sensor.consumption_sensor")
-    assert consumption_sensor.state == "0.002"
+    assert len(hass.states.async_all()) == 6
+    assert hass.states.get("sensor.clip_light_level_sensor").state == "999.8"
 
-    clip_light_level_sensor = hass.states.get("sensor.clip_light_level_sensor")
-    assert clip_light_level_sensor.state == "999.8"
+    # Disallow clip sensors
 
     hass.config_entries.async_update_entry(
         gateway.config_entry, options={deconz.gateway.CONF_ALLOW_CLIP_SENSOR: False}
     )
     await hass.async_block_till_done()
 
-    assert "sensor.light_level_sensor" in gateway.deconz_ids
-    assert "sensor.presence_sensor" not in gateway.deconz_ids
-    assert "sensor.switch_1" not in gateway.deconz_ids
-    assert "sensor.switch_1_battery_level" not in gateway.deconz_ids
-    assert "sensor.switch_2" not in gateway.deconz_ids
-    assert "sensor.switch_2_battery_level" in gateway.deconz_ids
-    assert "sensor.daylight_sensor" not in gateway.deconz_ids
-    assert "sensor.power_sensor" in gateway.deconz_ids
-    assert "sensor.consumption_sensor" in gateway.deconz_ids
-    assert "sensor.clip_light_level_sensor" not in gateway.deconz_ids
     assert len(hass.states.async_all()) == 5
+    assert hass.states.get("sensor.clip_light_level_sensor") is None
+
+    # Allow clip sensors
 
     hass.config_entries.async_update_entry(
         gateway.config_entry, options={deconz.gateway.CONF_ALLOW_CLIP_SENSOR: True}
     )
     await hass.async_block_till_done()
 
-    assert "sensor.light_level_sensor" in gateway.deconz_ids
-    assert "sensor.presence_sensor" not in gateway.deconz_ids
-    assert "sensor.switch_1" not in gateway.deconz_ids
-    assert "sensor.switch_1_battery_level" not in gateway.deconz_ids
-    assert "sensor.switch_2" not in gateway.deconz_ids
-    assert "sensor.switch_2_battery_level" in gateway.deconz_ids
-    assert "sensor.daylight_sensor" not in gateway.deconz_ids
-    assert "sensor.power_sensor" in gateway.deconz_ids
-    assert "sensor.consumption_sensor" in gateway.deconz_ids
-    assert "sensor.clip_light_level_sensor" in gateway.deconz_ids
     assert len(hass.states.async_all()) == 6
+    assert hass.states.get("sensor.clip_light_level_sensor")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6271139)